libs/src/svkImageReader2.h

```diff
@@ -65,20 +64,46 @@ class svkImageReader2 : public vtkImageReader2
 
         vtkTypeRevisionMacro( svkImageReader2, vtkImageReader2);
 
+        typedef enum {
+            UNDEFINED = -1,
+            DICOM_MRS=0,
+            DICOM_MRI,
+            DICOM_ENHANCED_MRI,
+            DICOM_SEGMENTATION,
+            DICOM_RAW,
+            SIEMENS_RDA,
+            VARIAN_FID,
+            VARIAN_FDF,
+            GE_PFILE,
+            GE_SDBM,
+            GE_POSTAGE_STAMP,
+            GE_SIGNA_LX2,
+            GE_SIGNA_5X,
+            LC_MODEL_COORD, 
+            LC_MODEL_CSV, 
+            BRUKER_MRS,
+            DDF,
+            IDF,
+            LAST_TYPE = IDF
+        } ReaderType;
+
```

Comment:
Nothing!  Just following the convention in the writers of using an enumeration.

__________________
Jason Crane, Ph.D.
Department of Radiology and Biomedical Imaging
UCSF Radiology MC 2532
Byers Hall Suite 301
CA Institute for Quantitative Biomedical Research
1700 4th St.<x-apple-data-detectors://6/0>
San Francisco, CA 94158-2330<x-apple-data-detectors://6/0>
e-mail: jason.crane@ucsf.edu<mailto:jasonc@radiology.ucsf.edu>
tel: 415.514.4426<tel:415.514.4426>
FAX: 415.514.2550<tel:415.514.2550>

On Dec 17, 2015, at 1:44 PM, Beck Olson <notifications@github.com<mailto:notifications@github.com>> wrote:


In libs/src/svkImageReader2.h<https://github.com/SIVICLab/sivic/pull/12#discussion_r47964572>:

> +            SIEMENS_RDA,
> +            VARIAN_FID,
> +            VARIAN_FDF,
> +            GE_PFILE,
> +            GE_SDBM,
> +            GE_POSTAGE_STAMP,
> +            GE_SIGNA_LX2,
> +            GE_SIGNA_5X,
> +            LC_MODEL_COORD,
> +            LC_MODEL_CSV,
> +            BRUKER_MRS,
> +            DDF,
> +            IDF,
> +            LAST_TYPE = IDF
> +        } ReaderType;
> +


What is the advantage of using enums over using the class name with GetClassName?

—
Reply to this email directly or view it on GitHub<https://github.com/SIVICLab/sivic/pull/12/files#r47964572>.